joss-paper/paper.md

```diff
@@ -0,0 +1,202 @@
+   index: 3
+ - name: Scuola Normale Superiore, Piazza dei Cavalieri 7, 56126 Pisa, Italy
+   index: 4
+ - name: Department of Physics, Harvard University, 17 Oxford St., Cambridge, MA, 02138, USA
+   index: 5
+date: 29 Feb 2020
+bibliography: paper.bib
+---
+
+# Summary
+
+The field of 21cm cosmology -- in which the hyperfine spectral line of neutral hydrogen
+(appearing at rest-frame wavelengths of 21cm) is mapped over large swathes of the
+Universe's history -- has developed radically over the last decade.
+Notwithstanding the overwhelming technical challenges associated with the observation
+and eventual detection of this signal, the promise of the field is to revolutionize our
+knowledge of the first stars, galaxies, and black holes through the timing and patterns
+they imprint in the cosmic 21cm signal.
+In order to interpret the eventual observational data, a range of physical models have
+been developed -- from simple analytic models of the global history of hydrogen reionization,
+through to fully hydrodynamical simulations of the 3D evolution of the brightness
+temperature of the spectral line.
+Between these extremes lies an especially versatile middle-ground: fast semi-numerical
+models that approximate the full 3D evolution of the relevant fields: density, velocity,
+temperature, ionization, and radiation.
+These have the advantage of being comparable to the full first-principles
+hydrodynamic simulations, but significantly quicker to run; so much so that they can
+be used to produce thousands of realizations on scales comparable to those observable
+by upcoming telescopes, in order to explore the very wide
+parameter space that still remains consistent with the data.
+
+
+Amongst practitioners in the field of 21cm cosmology, the `21cmFAST` program has become
+the *de facto* standard for such semi-numerical simulators.
+`21cmFAST` [@mesinger2007,@mesinger2010] is a high-performance C code that uses the
+excursion set formalism [@furlanetto2004] to
+identify regions of ionized hydrogen atop a cosmological density field evolved using
+first- or second-order Lagrangian perturbation theory [@zeldovich1970, @scocciomarro2002],
+tracking the thermal and ionization state of the intergalactic medium, and computing
+X-ray, soft UV and ionizing UV cosmic radiation fields based on parametrized galaxy models.
+For example, the following figure contains slices of lightcones (3D fields in which one
+axis corresponds to both spatial *and* temporal evolution) for the various
+component fields produced by `21cmFAST`.
+
+![Sample of Component Fields output by 21cmFAST](yuxiangs_big_plot_smaller.pdf){height=300px}
+
+However, `21cmFAST` is a highly specialized code, and its implementation has been
+quite specific and relatively inflexible.
+This inflexibility makes it hard to modify the behaviour of the code without detailed
+knowledge of the full system, and without disrupting the workings of the full system.
```

Comment:
full system is repeated, change the last one to code maybe?

Reply:
Thanks, changed to "This inflexibility makes it difficult to modify the behaviour of the code without detailed knowledge of the full system, or disrupting its workings."